### ISSUE TITLE
DateTime fields formatted inconsistently with CSV/XLSX/JSON downloads and email subscriptions

### ISSUE THREAD

**Issue by rognstad** [metabase/metabase#40093]
### Describe the bug

I have a dashboard that includes a question with two DateTime fields. They are configured like this:
<img width="293" alt="image" src="https://github.com/metabase/metabase/assets/6739933/60e68d11-a712-4183-9a16-5983305d5911">

When viewed within the Metabase application, they render as expected:
<img width="461" alt="image" src="https://github.com/metabase/metabase/assets/6739933/13877d0c-5f7d-4ff0-ac57-3911b4a4032c">

When downloading the card's data from the dashboard, Excel correctly renders the second column but not the first:
<img width="726" alt="image" src="https://github.com/metabase/metabase/assets/6739933/bbc1dcb4-14ae-401c-8a7a-1de02e184b3c">
The data is all there for the first column, though. If I manually change the formatting of the cells to include the time, then it displays as expected.

CSV and JSON download results in text that doesn't match my chosen format at all. Instead the DateTimes are formatted like  YYYY-MM-DDTHH:MM:SS:
<img width="375" alt="image" src="https://github.com/metabase/metabase/assets/6739933/7c59d858-a8eb-4335-95d4-cebae14ff326">

If instead of using the download card feature I setup a subscription for the dashboard, I get a 4th format in the email that only shows the date, no time information:
<img width="443" alt="image" src="https://github.com/metabase/metabase/assets/6739933/158ee755-9f66-4a8f-b316-442242e3106a">

If I look at the attachment to that email, I get a 5th format:
<img width="317" alt="image" src="https://github.com/metabase/metabase/assets/6739933/d487948d-72df-4a8b-ab59-952f34e9f651">
Like with CSV, it is using YYYY-MM-DDTHH:MM:SS, but it adds a Z at the end, indicating that this is in UTC (though it still shows the same hour as in all the other contexts, which have been localized to the user's timezone).

### To Reproduce

1. Format a DateTime field on a card on a dashboard
2. Download as CSV
3. Download as JSON
4. Download as Excel
5. Check the HTML body and attachment of an emailed subscription to the dashboard

You will see many different formats for the column

### Expected behavior

The text in all contexts should match the way it displays within the Metabase application (i.e. the way the column is configured)

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.22+7-post-Ubuntu-0ubuntu220.04.1",
    "java.vendor": "Ubuntu",
    "java.vendor.url": "https://ubuntu.com/",
    "java.version": "11.0.22",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.22+7-post-Ubuntu-0ubuntu220.04.1",
    "os.name": "Linux",
    "os.version": "5.15.0-1052-aws",
    "user.language": "en",
    "user.timezone": "Etc/UTC"
  },
  "metabase-info": {
    "databases": [
      "bigquery-cloud-sdk",
      "mysql",
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "13.13"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.6.0"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-03-04",
      "tag": "v1.48.8",
      "hash": "a900c85"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

It is merely annoying, but we're getting enough customer complaints that I need to update the cards to have columns formatted in SQL instead of relying on Metabase to do it

**Comment by paoliniluis** [comment]
I'm 95% sure this is fixed in v49 @rognstad

**Comment by adam-james-v** [comment]
@rognstad, this is fixed in 49.3 (or later).
